src/main/java/io/appium/java_client/remote/MobileOptions.java

```diff
@@ -512,4 +518,20 @@ protected T amend(String optionName, Object value) {
         setCapability(optionName, value);
         return (T) this;
     }
+
+    @Override
+    public Map<String, Object> asMap() {
+        Map<String, Object> toReturn = new TreeMap<>();
+
+        Map<String, Object> capsMap = super.asMap();
+        capsMap.forEach((key, value) -> {
+            if (key.equals("platformName")) {
```

Comment:
I think it is not only the platform name. W3C defines a bunch of keys, which must be supplied without prefixes and Selenium has this list somewhere in constants

Reply:
Hmm; `platformName` is the only one we specify in `MobileOptions`, but you are probably right here. Looks like `NewAppiumSessionPayload` uses `AcceptedW3CCapabilityKeys` which is probably the right fix?